sonic-xcvrd/scripts/xcvrd

```diff
@@ -0,0 +1,430 @@
+        return PHYSICAL_PORT_NOT_EXIST
+
+    if len(physical_port_list) > 1:
+        ganged_port = True
+
+    for physical_port in physical_port_list:
+        port_name = get_physical_port_name(logical_port_name, ganged_member_num, ganged_port)
+        ganged_member_num += 1
+
+        try:
+            dom_info_dict = platform_sfputil.get_transceiver_dom_info_dict(physical_port)
+            if dom_info_dict is not None:
+                beautify_dom_info_dict(dom_info_dict)
+                fvs = swsscommon.FieldValuePairs([('temperature', dom_info_dict['temperature']),
+                                                  ('voltage', dom_info_dict['voltage']),
+                                                  ('rx1power', dom_info_dict['rx1power']),
+                                                  ('rx2power', dom_info_dict['rx2power']),
+                                                  ('rx3power', dom_info_dict['rx3power']),
+                                                  ('rx4power', dom_info_dict['rx4power']),
+                                                  ('tx1bias', dom_info_dict['tx1bias']),
+                                                  ('tx2bias', dom_info_dict['tx2bias']),
+                                                  ('tx3bias', dom_info_dict['tx3bias']),
+                                                  ('tx4bias', dom_info_dict['tx4bias'])])
+                table.set(port_name, fvs)
+            else:
+                return SFP_EEPROM_NOT_READY
+
+        except NotImplementedError:
+            log_error("This functionality is currently not implemented for this platform")
+            sys.exit(3)
+
+# del sfp and dom info from db
+def del_port_sfp_dom_info_to_db(logical_port_name, int_tbl, dom_tbl):
+    ganged_port = False
+    ganged_member_num = 1
+
+    physical_port_list = logical_port_name_to_physical_port_list(logical_port_name)
+    if physical_port_list is None:
+        log_error("Error: No physical ports found for logical port '%s'" % logical_port_name)
+        return PHYSICAL_PORT_NOT_EXIST
+
+    if len(physical_port_list) > 1:
+        ganged_port = True
+
+    for physical_port in physical_port_list:
+        port_name = get_physical_port_name(logical_port_name, ganged_member_num, ganged_port)
+        ganged_member_num += 1
+
+        try:
+            int_tbl._del(port_name)
+            dom_tbl._del(port_name)
+
+        except NotImplementedError:
+            log_error("This functionality is currently not implemented for this platform")
+            sys.exit(3)
+
+# timer thread for dom info update
+def periodically_post_dom_info_to_db(table):
+    logical_port_list = platform_sfputil.logical
+    for logical_port_name in logical_port_list:
+        post_port_dom_info_to_db(logical_port_name, table)
+
+    global timer
+    timer = threading.Timer(DOM_INFO_UPDATE_PERIOD, periodically_post_dom_info_to_db, [table])
```

Comment:
>periodically_post_dom_info_to_db [](start = 52, length = 32)

This is weird recursion. Lots of timers will be created quickly 1->2->4->... ?

Reply:
No, if using a local variable 'timer' it will be the case as you said, here I am using a global 'timer' variable, so no this issue, timer triggered as expected.